src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -398,5 +408,17 @@ public static void ValidateCredentials(string username, string password, string
                 throw new CommandException(Strings.SourcesCommandCredentialsRequiredWithAuthTypes);
             }
         }
+
+        public static int ParseProtocolVersion(string protocolVersion)
+        {
+            var currentlySupportedProtocolVersions = new List<int> { 2, 3 };
```

Comment:
instead of hard coding protocol versions here, I would use PackageSource.DefaultProtocolVersion instead of 2. May be add a new variable named `MaxProtocolVersion` and then update the condition to accept a protocol version that falls within the the range of `DefaultProtocolVersion` to `MaxProtocolVersion`.

https://github.com/NuGet/NuGet.Client/blob/80c3c74ca5c6d67912a19833e8cc2b10f875dc95/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs#L19

Reply:
I've pushed code addressing this.

Reply:
@kartheekp-ms this conversation is not marked as resolved. Can you please check if you have any other comments?